### PR TITLE
Third branch

### PR DESCRIPTION
**Pull request by dubtouch**

### CODE REVIEW

File: MineSweeper/src/minesweeper/MineSweeper.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                } catch (IndexOutOfBoundsException e) {
+                }
+            }
+        }
+        if (counter == 0) {
+            field[i][j].setSymbol('/');
+            for (int a = i - 1; a < i + 2; a++) {
+                for (int b = j - 1; b < j + 2; b++) {
+                    try {
+                        if (!field[a][b].isMine() && field[a][b].getSymbol() == '.' || !field[a][b].isMine() && field[a][b].getSymbol() == '*') {
```
Comment (V-Negura) [inline MineSweeper/src/minesweeper/MineSweeper.java:112]: `field[a][b].getSymbol == '.'` should be a cell method as well as `field[a][b].getSymbol() == '*'`

File: MineSweeper/src/minesweeper/MineSweeper.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                    break;
+                }
+            }
+            if (hiddenCells == mines) break;
+        }
+        printField();
+        System.out.println("Congratulations! You found all mines!");
+    }
+
+    public void setField(int mines) {
```
Comment (V-Negura) [inline MineSweeper/src/minesweeper/MineSweeper.java:54]: Shouldn't this be private ?

File: MineSweeper/src/minesweeper/MineSweeper.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            int h = rand.nextInt(9);
+            int w = rand.nextInt(9);
+            if (!field[h][w].isMine()) {
+                field[h][w].setMine(true);
+                mines--;
+            }
+        }
+    }
+
+    public void printField() {
```
Comment (V-Negura) [inline MineSweeper/src/minesweeper/MineSweeper.java:71]: Shouldn't this be private ?

File: MineSweeper/src/minesweeper/MineSweeper.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            System.out.print(i + 1 + "|");
+            for (int j = 0; j < field[0].length; j++) {
+                System.out.print(field[i][j].getSymbol());
+            }
+            System.out.println("|");
+        }
+        System.out.println("-|---------|");
+    }
+
+    public void printFieldWithMines() {
```
Comment (V-Negura) [inline MineSweeper/src/minesweeper/MineSweeper.java:83]: Shouldn't this be private ?

File: MineSweeper/src/minesweeper/MineSweeper.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+            for (int j = 0; j < field[0].length; j++) {
+                if (field[i][j].isMine()) System.out.print('X');
+                else System.out.print(field[i][j].getSymbol());
+            }
+            System.out.println("|");
+        }
+        System.out.println("-|---------|");
+    }
+
+    public void checkCell(int i, int j) {
```
Comment (V-Negura) [inline MineSweeper/src/minesweeper/MineSweeper.java:96]: Shouldn't this be private ?

File: MineSweeper/src/minesweeper/MineSweeper.java [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+    public void setField(int mines) {
+        Random rand = new Random();
+        for (int i = 0; i < 9; i++) {
+            for (int j = 0; j < 9; j++) {
+                field[i][j] = new Cell('.', false);
+            }
+        }
+        while (mines > 0) {
+            int h = rand.nextInt(9);
```
Comment (V-Negura) [inline MineSweeper/src/minesweeper/MineSweeper.java:62]: Be more specify

File: MineSweeper/src/minesweeper/MineSweeper.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+            int w = rand.nextInt(9);
```
Comment (V-Negura) [inline MineSweeper/src/minesweeper/MineSweeper.java:63]: Be more specify

**Review by V-Negura — CHANGES_REQUESTED** [review]

File: MineSweeper/src/minesweeper/MineSweeper.java (hunk 1 above)
Comment (dubtouch) [reply MineSweeper/src/minesweeper/MineSweeper.java]: resolved

File: MineSweeper/src/minesweeper/MineSweeper.java (hunk 2 above)
Comment (dubtouch) [reply MineSweeper/src/minesweeper/MineSweeper.java]: resolved

File: MineSweeper/src/minesweeper/MineSweeper.java (hunk 3 above)
Comment (dubtouch) [reply MineSweeper/src/minesweeper/MineSweeper.java]: resolved

File: MineSweeper/src/minesweeper/MineSweeper.java (hunk 4 above)
Comment (dubtouch) [reply MineSweeper/src/minesweeper/MineSweeper.java]: resolved

File: MineSweeper/src/minesweeper/MineSweeper.java (hunk 5 above)
Comment (dubtouch) [reply MineSweeper/src/minesweeper/MineSweeper.java]: resolved

File: MineSweeper/src/minesweeper/MineSweeper.java (hunk 6 above)
Comment (dubtouch) [reply MineSweeper/src/minesweeper/MineSweeper.java]: resolved

File: MineSweeper/src/minesweeper/MineSweeper.java (hunk 7 above)
Comment (dubtouch) [reply MineSweeper/src/minesweeper/MineSweeper.java]: resolved

**Review by V-Negura — APPROVED** [review]

**Review by negura — APPROVED** [review]

**Pull request merged**